packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -34,12 +40,35 @@ public function __construct(string $definitionDirectory, string $indexPrefix, Cl
         $this->client = $client;
     }
 
+    /**
+     * @param int $buildVersion
+     * @internal Will be replaced with constructor injection in the next major release
+     */
+    public function setBuildVersion(int $buildVersion): void
+    {
+        if ($this->buildVersion !== null) {
+            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
+        }
+
+        $this->buildVersion = $buildVersion;
+    }
+
     /**
      * @param int $domainId
      * @param string $index
      * @return string
      */
     public function getIndexName(int $domainId, string $index): string
+    {
+        return $this->indexPrefix . $this->buildVersion . $index . $domainId;
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     * @return string
+     */
+    public function getAliasName(int $domainId, string $index): string
```

Comment:
Later, you call it "current" in some cases. Maybe renaming it to `getCurrentAliasName` could clarify things...